PowerEditor/src/WinControls/Grid/ShortcutMapper.cpp

```diff
@@ -352,6 +357,120 @@ INT_PTR CALLBACK ShortcutMapper::run_dlgProc(UINT message, WPARAM wParam, LPARAM
 					return TRUE;
 				}
 
+				case IDM_BABYGRID_CLEAR :
+				{
+					if (_babygrid.getNumberRows() < 1)
+						return TRUE;
+
+					NppParameters *nppParam = NppParameters::getInstance();
+					int row = _babygrid.getSelectedRow();
+					bool isModified = false;
+
+					switch(_currentState)
+					{
+						case STATE_MENU:
+						{
+							//Get CommandShortcut corresponding to row
+							vector<CommandShortcut> & shortcuts = nppParam->getUserShortcuts();
+							CommandShortcut csc = shortcuts[row - 1];
+							csc.clear();
+							shortcuts[row - 1] = csc;
+							//shortcut was altered
+							nppParam->addUserModifiedIndex(row-1);
+						
+							//save the current view
+							_lastHomeRow[_currentState] = _babygrid.getHomeRow();
+							_lastCursorRow[_currentState] = _babygrid.getSelectedRow();
+							fillOutBabyGrid();
+
+							isModified = true;
+
+							//Notify current Accelerator class to update everything
+							nppParam->getAccelerator()->updateShortcuts();
+							nppParam->setShortcutDirty();
+							break;
+						}

```

Comment:
There seems to be a good bit of code duplication among these `case` statements (and in other parts of the file as well). I'm not sure how easy it would be to pull out this common functionality into a separate function(s).